projects/etos_suite_runner/src/etos_suite_runner/lib/suite.py

```diff
@@ -93,36 +96,42 @@ def outcome(self) -> dict:
             return self.test_suite_finished.get("data", {}).get("testSuiteOutcome", {})
         return {}
 
-    def start(self, identifier: str) -> None:
+    def start(self, identifier: str, otel_context: opentelemetry.context.context.Context) -> None:
         """Start ETR for this sub suite.
 
         :param identifier: An identifier for logs in this sub suite.
         """
-        FORMAT_CONFIG.identifier = identifier
-        self.logger.info("Starting up the ETOS test runner", extra={"user_log": True})
-        executor = Executor(self.etos)
-        try:
-            executor.run_tests(self.environment)
-        except TestStartException as exception:
-            self.failed = True
-            self.logger.error(
-                "Failed to start sub suite: %s", exception.error, extra={"user_log": True}
-            )
-            raise
-        self.logger.info("ETR triggered.")
-        timeout = time.time() + self.etos.debug.default_test_result_timeout
-        try:
-            while time.time() < timeout:
-                time.sleep(10)
-                if not self.started:
-                    continue
-                self.logger.info("ETOS test runner has started", extra={"user_log": True})
-                self.request_finished_event()
-                if self.finished:
-                    self.logger.info("ETOS test runner has finished", extra={"user_log": True})
-                    break
-        finally:
-            self.release(identifier)
+        # OpenTelemetry context needs to be retrieved here:
+        # the subsuite is running in a separate process
+        span_name = "execute_testrunner"
+        with self.otel_tracer.start_as_current_span(span_name, context=otel_context) as span:
+            span.set_attribute("subsuite_id", identifier)
+            FORMAT_CONFIG.identifier = identifier
+            self.logger.info("Starting up the ETOS test runner", extra={"user_log": True})
+            executor = Executor(self.etos)
+            try:
+                executor.run_tests(self.environment)
+            except TestStartException as exception:
+                self.failed = True
+                self.logger.error(
+                    "Failed to start sub suite: %s", exception.error, extra={"user_log": True}
+                )
+                span.set_attribute("failed", str(exception))
+                raise
+            self.logger.info("ETR triggered.")
+            timeout = time.time() + self.etos.debug.default_test_result_timeout
+            try:
+                while time.time() < timeout:
+                    time.sleep(10)
```

Comment:
Isn't doing mandatory sleeping 10 a bit suboptimal? Couldn't we just have shorter sleep inside the 'not self.started'-loop?

Reply:
I'd avoid changing it here. This is old code, I just had to change the indent due to opentelemetry span recording.

Reply:
That sleep is there so that we do not spam the event repository too much. It is needed for both the `if not self.started` and further down if `self.finished` is False.
This loop is for waiting for the test runner to start and then to finish.